### PR TITLE
increase feemarket's consensus version and add migration handler

### PR DESCRIPTION
**Pull request by yihuang**
Closes: #1001

## Description

Add the migration handler to remove the unused state key.

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: x/feemarket/keeper/migrations.go [hunk 1]
```diff
@@ -11,3 +17,15 @@ func NewMigrator(keeper Keeper) Migrator {
 		keeper: keeper,
 	}
 }
+
+func (m Migrator) Migrate1to2(ctx sdk.Context) error {
+	store := ctx.KVStore(m.keeper.storeKey)
+	baseFeeKeyPrefix := []byte{2}
```
Comment (fedekunze) [inline x/feemarket/keeper/migrations.go:23]: use constants from types

File: x/feemarket/keeper/migrations.go [hunk 2]
```diff
@@ -11,3 +17,15 @@ func NewMigrator(keeper Keeper) Migrator {
...
+	bz := store.Get(baseFeeKeyPrefix)
+	if len(bz) > 0 {
```
Comment (fedekunze) [inline x/feemarket/keeper/migrations.go:25]: I'd also add a check if the paramset key is register

**Review by fedekunze — CHANGES_REQUESTED** [review]
LGTM, Can you add tests for the migration?

File: x/feemarket/keeper/migrations.go (hunk 1 above)
Comment (yihuang) [reply x/feemarket/keeper/migrations.go:23]: the constant is removed from types already.

File: x/feemarket/keeper/migrations.go (hunk 2 above)
Comment (yihuang) [reply x/feemarket/keeper/migrations.go:25]: what do you mean by registered, I think the paramset items are set directly?

**Comment by yihuang** [comment]
> LGTM, Can you add tests for the migration?

done, added a unit test.

**Review by fedekunze — APPROVED** [review]

**Pull request merged**